flash/core/model.py

```diff
@@ -91,6 +92,17 @@ def test_step(self, batch: Any, batch_idx: int) -> None:
         output = self.step(batch, batch_idx)
         self.log_dict({f"test_{k}": v for k, v in output["logs"].items()}, on_step=False, on_epoch=True, prog_bar=True)
 
+    @property
+    @contextmanager
+    def predict_context(self):
+        try:
+            self.eval()
+            torch.set_grad_enabled(False)
+            yield
+        finally:
+            self.train()
+            torch.set_grad_enabled(True)
```

Comment:
why the `try...finally`?

```suggestion
        self.eval()
        torch.set_grad_enabled(False)
        yield
        self.train()
        torch.set_grad_enabled(True)
```

Reply:
Moved to decorator.